src/server/rest/v1/service/PricingService.ts

```diff
@@ -59,13 +62,15 @@ export default class PricingService {
         onlyRecordCount: filteredRequest.OnlyRecordCount
       },
       authorizationPricingDefinitionsFilter.projectFields
-    );
+    ) as PricingDefinitionDataResult;
     // Assign projected fields
     if (authorizationPricingDefinitionsFilter.projectFields) {
       pricingDefinitions.projectFields = authorizationPricingDefinitionsFilter.projectFields;
     }
     // Add Auth flags
-    await AuthorizationService.addPricingDefinitionsAuthorizations(req.tenant, req.user, pricingDefinitions as PricingDataResult, authorizationPricingDefinitionsFilter);
+    await AuthorizationService.addPricingDefinitionsAuthorizations(req.tenant, req.user, pricingDefinitions , authorizationPricingDefinitionsFilter);
+    // Alter the canCreate flag according to the pricing definition context
+    pricingDefinitions.canCreate = await PricingService.alterCanCreate(req, action, filteredRequest.EntityType, filteredRequest.EntityID, pricingDefinitions.canCreate);
```

Comment:
pricingDefinitions.canCreate  should be set in addPricingDefinitionsAuthorizations()